src/Tests/dotnet-MsiInstallation.Tests/README.md

```diff
@@ -0,0 +1,57 @@
+
+## Setting up a VM for running tests
+
+The main requirements for running the tests are:
+
+- A Hyper-V Virtual Machine
+- Access to the VM via the admin share (`\\TestVM\c$`)
+- [PsExec](https://learn.microsoft.com/en-us/sysinternals/downloads/psexec) on the PATH of the host machine
+- "Remote Service Management" enabled in firewall settings inside the VM (this avoids a delay of 15-30 seconds for each command that is run)
+- Tests need to be run with admin priviledges, which are required to use the WMI APIs to control VMs
+
+Detailed steps:
+
+- [Enable Hyper-V](https://learn.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/enable-hyper-v)
+- For the tests to remotely control the VM, your host computer needs to be able to access the VM over the network.  To enable this, you need to create a virtual switch for the VM which will also be shared by the host PC.  In Hyper-V Manager, go to Virtual Switch Manager.  Create a new Virtual Switch connected to your external network adapter, and check the box that says "Allow management operating system to share this network adapter."
+- Create a Hyper-V Virtual machine
+  - You will need to choose a name for the virtual machine (used by the host) and a machine name for it when setting up Windows inside the VM.  You can choose whatever names you want, but "Test VM" and "TestVM" are good defaults.
+  - You can download a Windows 11 .iso here: https://www.microsoft.com/en-us/software-download/windows11/
+  - In the networking configuration for the VM, select the Virtual Switch you created
+  - Under Security, check "Enable Trusted Platform Module" (and possibly the "Encrypt state..." checkbox under it), which is required to install Windows 11
+  - Start the VM and install Windows
+    - Probably you don't want to sign on to the test VM with a Microsoft account.  Setting up with a local account is tricky, but you can do so with these steps: https://www.tomshardware.com/how-to/install-windows-11-without-microsoft-account
+  - In the VM settings in Hyper-V manager, enable all the integration services (so you can copy/paste files to the VM, for example)
+- In network settings inside the VM, switch the network connection type to Private Network, and turn on Network discovery and File and printer sharing.
+- Inside the VM, set the HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\System\LocalAccountTokenFilterPolicy registry value to 1 ([reference](https://learn.microsoft.com/en-us/troubleshoot/windows-server/windows-security/user-account-control-and-remote-restriction)).  This will allow you to access the admin share (`\\TestVM\c$`).
```

Comment:
```suggestion
- Inside the VM, set the HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\System\LocalAccountTokenFilterPolicy registry value to 1 ([reference](https://learn.microsoft.com/en-us/troubleshoot/windows-server/windows-security/user-account-control-and-remote-restriction)).  This will allow you to access the admin share (`\\TestVM\c$`).

You may also run this under an admin prompt:
`REG ADD HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Policies\System /v LocalAccountTokenFilterPolicy /t REG_DWORD /d 1`
```